src/game/enemy/systems.rs

```diff
@@ -201,4 +304,62 @@ pub fn catch_shot_event(
             }
         }
     }
-}
\ No newline at end of file
+}
+
+/// Enemy changes sprite sheet to shooting and triggers shot event
+pub fn enemy_shoot_player(
+    mut commands: Commands,
+    mut enemy_q: Query<(Entity, &Transform), (With<Enemy>, Without<Firing>)>,
+    enemy_handles: Res<EnemyHandles>,
+    mut enemy_shot_player_event_writer: EventWriter<EnemyShotPlayerEvent>,
+    keyboard_input: Res<Input<KeyCode>>, // Remove later
+) {
+    if keyboard_input.just_pressed(KeyCode::F2) {
+        println!("<< shoot >>");
+        let mut rng = rand::thread_rng();
+        if let Some((enemy_entity, transform)) = enemy_q.iter_mut().choose(&mut rng) {
+            // inserting this replaces the old one
+            commands.entity(enemy_entity).insert((
+                SpriteSheetBundle {
+                    texture_atlas: enemy_handles.soviet_fire.clone(),
+                    sprite: TextureAtlasSprite::new(0),
+                    transform: *transform,
+                    ..default()
+                },
+                EnemyAnimations::SovietFire.get_indices(),
+                EnemyAnimations::SovietFire.get_timer(),
+                Firing::default(),
+            ));
+            enemy_shot_player_event_writer.send(EnemyShotPlayerEvent(enemy_entity.index()));
+        }
+    }
+}
+
+/// Ticks enemy Firing timer until finished
+///
+/// When finished revert to base animation
+pub fn tick_and_replace_enemy_fire_timer(
+    mut commands: Commands,
+    mut firing_q: Query<(Entity, &mut Firing, &Transform), (With<Firing>, With<Enemy>)>,
+    enemy_handles: Res<EnemyHandles>,
+    time: Res<Time>,
+) {
+    for (enemy_entity, mut firing, transform) in firing_q.iter_mut() {
+        if firing.timer.just_finished() {
+            commands.entity(enemy_entity).remove::<Firing>();
+            // inserting this replaces the old one
+            commands.entity(enemy_entity).insert((
+                SpriteSheetBundle {
+                    texture_atlas: enemy_handles.soviet_idle.clone(),
+                    sprite: TextureAtlasSprite::new(0),
+                    transform: *transform,
+                    ..default()
+                },
+                EnemyAnimations::SovietIdle.get_indices(),
+                EnemyAnimations::SovietIdle.get_timer(),
+            ));
+        } else {
+            firing.timer.tick(time.delta());
```

Comment:
couldnt we tick this and then check if it just_finished in the same run of the method?